clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts

```diff
@@ -84,26 +27,24 @@ export function getURIFromResource(resource: string): urlUtil.URL[] {
  * @param copybookFolders list of folders from where to search the copybook
  * @param extensions list of possible copybooks extensions
  */
-export function searchCopybookInWorkspace(
+export function searchCopybookInExtensionFolder(
   copybookName: string,
   copybookFolders: string[] | undefined,
   extensions: string[] | undefined,
 ): string | undefined {
-  for (const workspaceFolderPath of SettingsUtils.getWorkspaceFoldersPath()) {
-    if (!copybookFolders || !extensions) return undefined;
-    const workspaceFolder = cleanWorkspaceFolder(workspaceFolderPath);
-    for (const p of copybookFolders) {
-      for (const ext of extensions) {
-        const searchResult = globSearch(workspaceFolder, p, copybookName, ext);
-        if (searchResult) {
-          const root = path.parse(searchResult).root;
-          const urlPath = searchResult
-            .substring(root.length)
-            .split(path.sep)
-            .map((s) => encodeURIComponent(s))
-            .join(path.sep);
-          return new urlUtil.URL("file://" + root + urlPath).href;
-        }
+  if (!copybookFolders || !extensions) return undefined;
+  const extensionFolder = cleanWorkspaceFolder(Utils.getExtensionFolder());
```

Comment:
`cleanWorkspaceFolder` should be renamed, also why not store the the clean workspace folder name?

Reply:
It is still being used for workspace folders here : https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/development/clients/cobol-lsp-vscode-extension/src/services/ProcessorGroups.ts#L43

Here since extension folder being used I think we don't need to clean folder name.

Reply:
That's why I said renamed - it is not cleaning workspace folder, at best it is manipulating its name.

Reply:
resolved at : 224e9b8e00a6753439dd54d8da5d6df093b6cf89